spec/requests/sidebar_sections_controller_spec.rb

```diff
@@ -334,5 +349,41 @@
 
       expect(response.status).to eq(403)
     end
+
+    it "doesn't allow moderator to delete public sidebar section" do
+      sign_in(moderator)
+      sidebar_section.update!(public: true)
+      delete "/sidebar_sections/#{sidebar_section.id}.json"
+
+      expect(response.status).to eq(403)
+    end
+  end
+
+  describe "#reset" do
+    let(:community_section) do
+      SidebarSection.find_by(section_type: SidebarSection.section_types[:community])
+    end
+    let(:everything_link) { community_section.sidebar_section_links.first }
+
+    it "doesn't allow user to reset community section" do
+      sign_in(user)
+      SidebarSection.any_instance.expects(:reset_community!).never
+      put "/sidebar_sections/reset/#{community_section.id}.json"
+      expect(response.status).to eq(403)
+    end
+
+    it "doesn't allow staff to reset community section" do
+      sign_in(moderator)
+      SidebarSection.any_instance.expects(:reset_community!).never
+      put "/sidebar_sections/reset/#{community_section.id}.json"
+      expect(response.status).to eq(403)
+    end
+
+    it "allows admins to reset community section to default" do
+      sign_in(admin)
+      SidebarSection.any_instance.expects(:reset_community!).once
+      put "/sidebar_sections/reset/#{community_section.id}.json"
+      expect(response.status).to eq(200)
```

Comment:
Technically, we also want to test what the response body is. Otherwise, the controller can just start returning nothing and our API will break.